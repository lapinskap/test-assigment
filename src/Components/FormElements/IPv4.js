import PropTypes from 'prop-types';
import React from 'react';
import {
  InputGroup,
  InputGroupAddon,
  FormGroup,
  Label,
  InputGroupText,
  Input, Row, Col,

} from 'reactstrap';
import ValidationMessage from '../Form/ValidationMessage';
import hashInputId from '../../utils/jsHelpers/hashInputId';

export default function IPv4({
  id, label, validation, placeholder, value, prefix, suffix, validateField, tooltip,
  errorMessage, onChange, tooltipPosition, inputSwitcher, disabled,
}) {
  const input = (
    <Input
      data-t1={id}
      type="text"
      minLength="7"
      maxLength="15"
      onChange={onChange}
      id={hashInputId(id)}
      name={hashInputId(id)}
      disabled={disabled}
      autoComplete="chrome-off"
      size="15"
      placeholder={placeholder}
      invalid={Boolean(errorMessage)}
      onBlur={(e) => {
        validateField(id, e.target.value, validation);
      }}
      value={value || ''}
    />
  );
  const inputGroup = (
    <InputGroup>
      {prefix ? (
        <InputGroupAddon addonType="prepend">
          <InputGroupText>{prefix}</InputGroupText>
        </InputGroupAddon>
      ) : null}
      {input}
      {suffix ? (
        <InputGroupAddon addonType="append">
          <InputGroupText>{suffix}</InputGroupText>
        </InputGroupAddon>
      ) : null}
      {tooltip && tooltipPosition === 'input' ? (
        <InputGroupAddon addonType="append">
          <InputGroupText>{tooltip}</InputGroupText>
        </InputGroupAddon>
      ) : null}
      <ValidationMessage message={errorMessage} />
    </InputGroup>
  );

  return (
    <FormGroup>
      <Label data-t1={`${id}Label`} for={id}>
        {label}
        {tooltip && tooltipPosition === 'label' ? (
          <>
            {' '}
            {tooltip}
          </>
        ) : null}
      </Label>
      {inputSwitcher ? (
        <Row>
          <Col sm={9}>
            {inputGroup}
          </Col>
          <Col sm={3}>
            {inputSwitcher}
          </Col>
        </Row>
      ) : inputGroup}
    </FormGroup>
  );
}

IPv4.propTypes = {
  id: PropTypes.string,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]),
  onChange: PropTypes.func.isRequired,
  tooltip: PropTypes.node,
  inputSwitcher: PropTypes.node,
  disabled: PropTypes.bool,
  tooltipPosition: PropTypes.string,
  value: PropTypes.string,
  errorMessage: PropTypes.string,
  validateField: PropTypes.func,
  placeholder: PropTypes.string,
  prefix: PropTypes.string,
  suffix: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  validation: PropTypes.array,
};

IPv4.defaultProps = {
  id: '',
  label: '',
  tooltip: null,
  tooltipPosition: 'label',
  inputSwitcher: null,
  disabled: false,
  value: '',
  errorMessage: null,
  validateField: null,
  validation: null,
  placeholder: '',
  prefix: '',
  suffix: '',
};
