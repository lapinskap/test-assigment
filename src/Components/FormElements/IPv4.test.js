import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import IPv4 from './IPv4';
import '../../tests/setupTests';

describe('<IPv4 />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  let id; let label; let validation; let isInValid;
  let placeholder; let value; let prefix; let suffix; let validateField; let tooltip; let errorMessage; let
    onChange;

  beforeEach(() => {
    id = 1;
    label = 'test';
    validation = ['required'];
    isInValid = false;
    placeholder = '';
    value = '';
    prefix = '';
    suffix = '';
    validateField = '';
    tooltip = '';
    errorMessage = '';
    onChange = () => ({});
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all props', () => {
    const tree = renderer
      .create(<IPv4
        id={id}
        label={label}
        validation={validation}
        isInValid={isInValid}
        placeholder={placeholder}
        value={value}
        prefix={prefix}
        suffix={suffix}
        validateField={validateField}
        tooltip={tooltip}
        errorMessage={errorMessage}
        onChange={onChange}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without optional props', () => {
    const tree = renderer
      .create(<IPv4
        onChange={onChange}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
