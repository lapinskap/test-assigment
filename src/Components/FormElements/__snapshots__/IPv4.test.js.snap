// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IPv4 /> renders correctly with all props 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    data-t1="1Label"
    htmlFor={1}
  >
    test
  </label>
  <div
    className="input-group"
  >
    <input
      aria-invalid={false}
      autoComplete="chrome-off"
      className="form-control"
      data-t1={1}
      disabled={false}
      id={1}
      maxLength="15"
      minLength="7"
      name={1}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      size="15"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<IPv4 /> renders correctly without optional props 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    data-t1="Label"
    htmlFor=""
  >
    
  </label>
  <div
    className="input-group"
  >
    <input
      aria-invalid={false}
      autoComplete="chrome-off"
      className="form-control"
      data-t1=""
      disabled={false}
      id=""
      maxLength="15"
      minLength="7"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      size="15"
      type="text"
      value=""
    />
  </div>
</div>
`;
